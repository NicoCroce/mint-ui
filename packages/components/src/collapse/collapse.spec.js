import React from "react";
import {  render, fireEvent, cleanup, getByTestId, queryAllByTestId } from "@testing-library/react";
import Collapse, { Panel } from './index';

afterEach(cleanup)

describe("collapse", () => {
    test("renders correctly", () => {
        const { asFragment } = render(
            <Collapse accordion={true} expandIconPosition="right" activeKey={2}>
                <Panel panelKey={1} header="Panel Header 1">
                    Some content
                </Panel>
                <Panel panelKey={2} header="Panel Header 2">
                    Some content
                </Panel>
                <Panel panelKey={3} header="Panel Header 3" disabled>
                    Some content
                </Panel>
            </Collapse>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    test("toggles", () => {
        const { queryAllByTestId, queryByTestId, getByTestId } = render(
            <Collapse data-testid="collapse" activeKey={2} onChange={key => { console.log('key', key) }}>
                <Panel panelKey={1} header="Panel Header 1">
                    Some content 1
                </Panel>
                <Panel panelKey={2} header="Panel Header 2">
                    Some content 2
                </Panel>
                <Panel panelKey={3} header="Panel Header 3" disabled>
                    Some content 3
                </Panel>
            </Collapse>
        );
        expect(queryAllByTestId("tm-collapse-body").length).toBe(1);
        fireEvent.click(queryAllByTestId("tm-panel-header")[0]);
        expect(queryAllByTestId("tm-collapse-body").length).toBe(2);
    });
});