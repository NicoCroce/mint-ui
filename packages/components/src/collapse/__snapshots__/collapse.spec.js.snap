// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapse renders correctly 1`] = `
<DocumentFragment>
  <div
    class="collapse__CollapseWrap-sc-1gbzbz3-0 jScgZA"
  >
    <div
      class="tm-panel"
    >
      <div
        class="collapse__Header-sc-1gbzbz3-2 bIOJjx tm-panel-header"
        data-testid="tm-panel-header"
      >
        <div
          style="display: inline-flex;"
        >
          <span>
            Panel Header 1
          </span>
        </div>
        <div
          style="display: inline-flex;"
        >
          <i
            class="collapse__StyledEL-sc-1gbzbz3-1 gWQSeT"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="#a1a6a7"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="tm-panel"
    >
      <div
        class="collapse__Header-sc-1gbzbz3-2 bIOJjx tm-panel-header"
        data-testid="tm-panel-header"
      >
        <div
          style="display: inline-flex;"
        >
          <span>
            Panel Header 2
          </span>
        </div>
        <div
          style="display: inline-flex;"
        >
          <i
            class="collapse__StyledEL-sc-1gbzbz3-1 gWQSeT"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                fill="#a1a6a7"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </i>
        </div>
      </div>
      <div
        class="collapse__Body-sc-1gbzbz3-3 diwbzO tm-collapse-body"
        data-testid="tm-collapse-body"
      >
        Some content
      </div>
    </div>
    <div
      class="tm-panel"
    >
      <div
        class="collapse__Header-sc-1gbzbz3-2 fbesdd tm-panel-header"
        data-testid="tm-panel-header"
        disabled=""
      >
        <div
          style="display: inline-flex;"
        >
          <span>
            Panel Header 3
          </span>
        </div>
        <div
          style="display: inline-flex;"
        >
          <i
            class="collapse__StyledEL-sc-1gbzbz3-1 gWQSeT"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="#a1a6a7"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
